pubs/repo.py

```diff
@@ -192,8 +192,17 @@ def push_doc(self, citekey, docfile, copy=None):
         p.docpath = docfile
         self.push_paper(p, overwrite=True, event=False)
 
-    def unique_citekey(self, base_key):
-        """Create a unique citekey for a given basekey."""
+    def unique_citekey(self, base_key, bibentry):
+        """Create a unique citekey for a given base key.
+
+        :param base_key:  the base key in question.
+        :param bibentry:  the bib entry to possibly generate the citekey.
+        """
+        # can't have `/` in citekeys
+        # FIXME: a bit crude, but efficient for now (and allows unicode citekeys)
+        # TODO:  check that the generated citekey does not have a slash too.
+        if '/' in base_key:
```

Comment:
Maybe we could export that into a `is_citekey_valid` helper function since it will likely be used elsewhere. There is already a TODO in rename for validating citekeys and I guess this won't be the last one!